src/components/Share.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import {shareGoogleDrive} from '../redux/actionCreators';
+import {bindActionCreators} from 'redux';
+import {styles} from './styles';
+import Button from './Button';
+import Spinner from './Spinner';
+import * as selectors from '../redux/selectors';
+
+export class Share extends React.Component {
+  render() {
+    return (
+      <span>
+        {this.props.hasSavedDrive && !this.props.isSharingDrive &&
+          <Button kind="toolbar"
+                  onClick={() => this.props.shareGoogleDrive()}>
+            Share
+          </Button>
+        }
+        {this.props.hasSavedDrive && this.props.isSharingDrive &&
+          <Button kind="toolbar">
+            <Spinner style={styles.spinners.toolbar}/>
+            Sharing...
+          </Button>
+        }
+      </span>
+    );
+  }
+}
+
+Share.propTypes = {
+  hasSavedDrive: React.PropTypes.any,
+  isSharingDrive: React.PropTypes.bool,
+  hasSharedDrive: React.PropTypes.any,
```

Comment:
`state.googleDrive.save === !null` is invalid syntax. You want `state.googleDrive.save !== null`.

In any case, if the component is going to treat the value like a bool, then you should make the prop type be a bool. If the selector is returning something other than a bool, then you should cast it to a bool when you connect the component to the redux state.